[TraceQL] Add API SearchTagValues V2 (#1956)
grafana/tempo: docs/sources/api_docs/_index.md

}
```

### Search tag values V2

This endpoint retrieves all discovered values and their data types for the given TraceQL identifier.  The endpoint is available in the query frontend service in

> a microservices deployment, or the Tempo endpoint in a monolithic mode deployment. This endpoint is similar to `/api/search/tag/<tag>/values` but operates on TraceQL identifiers and types.  See [TraceQL](../traceql/) documention for more information. The following request will return all discovered service names.

Reviewer: Clear explanation of what this does and it has link! Best practice to avoid using future tense, when possible. 
```suggestion
a microservices deployment, or the Tempo endpoint in a monolithic mode deployment. This endpoint is similar to `/api/search/tag/<tag>/values` but operates on TraceQL identifiers and types. See [TraceQL](../traceql/) documention for more information. The following request returns all discovered service names.
```

```
GET /api/search/tag/.service.name/values
```

#### Example

Example of how to query Tempo using curl.

> This query will return all discovered values for the tag `service.name`.

Reviewer: ```suggestion
This query returns all discovered values for the tag `service.name`.
```

> Example of how to query Tempo using curl.
> This query will return all discovered values for the tag `service.name`.

Reviewer: ```suggestion
This example queries Tempo using curl and returns all discovered values for the tag `service.name`.
```
